src/ui/components/PasswordModule/PasswordModule.test.tsx

```diff
@@ -0,0 +1,623 @@
+      ionFireEvent.click(submitButton);
+    });
+
+    await waitFor(() => {
+      expect(secureStorageSetFunc).toBeCalledWith(
+        KeyStoreKeys.APP_OP_PASSWORD,
+        "Passssssssss1@"
+      );
+      expect(createOrUpdateBasicRecordMock).toBeCalledWith(
+        new BasicRecord({
+          id: MiscRecordId.OP_PASS_HINT,
+          content: { value: "hint" },
+        })
+      );
+    });
+
+    spy.mockRestore();
+  });
+
+  test("Submit password on manage password page", async () => {
+    const initialState = {
+      stateCache: {
+        routes: [RoutePath.GENERATE_SEED_PHRASE],
+        authentication: {
+          loggedIn: true,
+          time: Date.now(),
+          passcodeIsSet: true,
+          seedPhraseIsSet: true,
+          passwordIsSet: true,
+        },
+        currentOperation: OperationType.IDLE,
+      },
+      seedPhraseCache: {
+        seedPhrase: "",
+        bran: "",
+      },
+      cryptoAccountsCache: {
+        cryptoAccounts: [],
+      },
+      biometricsCache: {
+        enabled: false,
+      },
+    };
+
+    const { getByTestId, queryByText, getByText } = render(
+      <Provider store={storeMocked(initialState)}>
+        <PasswordModule
+          title="Password Module"
+          description="Description"
+          testId="password-module"
+          isOnboarding={false}
+          onCreateSuccess={onCreateSuccesMock}
+        />
+      </Provider>
+    );
+
+    expect(queryByText(ENG_trans.createpassword.button.skip)).toBe(null);
+
+    const input = getByTestId("create-password-input");
+    const confirmInput = getByTestId("confirm-password-input");
+    const hintInput = getByTestId("create-hint-input");
+
+    act(() => {
+      ionFireEvent.ionInput(input, "Passssssss1@");
+      ionFireEvent.ionInput(confirmInput, "Passssssss1@");
+      ionFireEvent.ionInput(hintInput, "hint");
+    });
+
+    await waitFor(() => {
+      expect(
+        getByTestId("password-validation-length-icon").classList.contains(
+          "pass"
+        )
+      ).toBe(true);
+      expect(
+        getByTestId("password-validation-uppercase-icon").classList.contains(
+          "pass"
+        )
+      ).toBe(true);
+      expect(
+        getByTestId("password-validation-lowercase-icon").classList.contains(
+          "pass"
+        )
+      ).toBe(true);
+      expect(
+        getByTestId("password-validation-number-icon").classList.contains(
+          "pass"
+        )
+      ).toBe(true);
+      expect(
+        getByTestId("password-validation-symbol-icon").classList.contains(
+          "pass"
+        )
+      ).toBe(true);
+    });
+
+    const submitButton = getByTestId("primary-button-password-module");
+
+    act(() => {
+      ionFireEvent.click(submitButton);
+    });
+
+    await waitFor(() => {
+      expect(
+        getByText(
+          ENG_trans.settings.sections.security.managepassword.page.alert
+            .existingpassword
+        )
+      ).toBeVisible();
+    });
```

Comment:
Would you be able to capture `actionConfirm={() => handleClearExisting()}` when we click on "Ok"?